### PR TITLE
Documentation cleanup in std

### PR DESCRIPTION
**Pull request by Aurel300**
- [x] Remove doc comments in target-specific `_std` directories. In most cases they were exact copies of the extern version. In some cases they were stale or slightly different. In other cases there were some target-specific comments, which were moved to the extern doc.

Stay tuned for the next PR:

 - [ ] Fix a whole bunch of grammatical and stylistic (Markdown-breaking) issues.
 - [ ] More descriptive argument names.

### CODE REVIEW

File: std/haxe/ds/IntMap.hx [hunk 1]
```diff
@@ -57,11 +57,17 @@ extern class IntMap<T> implements haxe.Constraints.IMap<Int, T> {
 
 	/**
 		See `Map.keys`
+
+		(cs, java) Implementation detail: Do not `set()` any new value while
+		iterating, as it may cause a resize, which will break iteration.
```
Comment (Simn) [inline std/haxe/ds/IntMap.hx:62]: Note that modify-while-iterate is undefined behavior in general.

File: std/sys/thread/Tls.hx [hunk 2]
```diff
@@ -28,7 +28,10 @@ package sys.thread;
 
 /**
 	Creates thread local storage.
- */
+
+	(hl) Warning: At the moment `Tls` does not protect the value from being
```
Comment (Simn) [inline std/sys/thread/Tls.hx:32]: That sounds like something we should fix, not document...

**Comment by Simn** [comment]
For reviewing purposes, I'm in favor of merging this as-is so the diff is smaller for the parts that actually change content.

File: std/haxe/ds/IntMap.hx (hunk 1 above)
Comment (Aurel300) [reply std/haxe/ds/IntMap.hx:62]: Are there any tests for this? Are there any targets where it actually works fine?

**Comment by Aurel300** [comment]
I agree, I'll make a separate PR for the other changes.

File: std/sys/thread/Tls.hx (hunk 2 above)
Comment (Aurel300) [reply std/sys/thread/Tls.hx:32]: https://github.com/HaxeFoundation/hashlink/issues/284

**Pull request merged**